rpcclient/infrastructure.go

```diff
@@ -1534,3 +1620,60 @@ func (c *Client) BackendVersion() (BackendVersion, error) {
 
 	return *c.backendVersion, nil
 }
+
+func (c *Client) sendAsync() FutureGetBulkResult {
+	// convert the array of marshalled json requests to a single request we can send
+	responseChan := make(chan *response, 1)
+	marshalledRequest := []byte("[")
+	for iter := c.batchList.Front(); iter != nil; iter = iter.Next() {
+		request := iter.Value.(*jsonRequest)
+		marshalledRequest = append(marshalledRequest, request.marshalledJSON...)
+		marshalledRequest = append(marshalledRequest, []byte(",")...)
+	}
+	if len(marshalledRequest) > 0 {
+		// removes the trailing comma to process the request individually
+		marshalledRequest = marshalledRequest[:len(marshalledRequest)-1]
+	}
+	marshalledRequest = append(marshalledRequest, []byte("]")...)
+	request := jsonRequest{
+		id:             c.NextID(),
+		method:         "",
+		cmd:            nil,
+		marshalledJSON: marshalledRequest,
+		responseChan:   responseChan,
+	}
+	c.sendPost(&request)
+	return responseChan
+}
+
+// Marshall's bulk requests and sends to the server
+// creates a response channel to receive the response
+func (c *Client) Send() error {
+	result, err := c.sendAsync().Receive()
+
```

Comment:
I think we need to do something like this here, in order to clear the `batchList` even in case of an error:

```
defer func() {
    c.batchList = list.New()
}
```

What do you think?

Reply:
If `Send()` is called on an empty `batchList`, it should be a noop.

```
if c.batchList.Len() == 0 {
    return nil
}
```

The above needs to be called **before** `result, err := c.sendAsync().Receive()` is done to prevent a JSON marshalling error.